fala/apps/adviser/tests/test_crown_dependencies_view.py

```diff
@@ -8,33 +8,28 @@ class PostcodeValidationTest(SimpleTestCase):
     client = Client()
     url = reverse("search")
 
-    def test_guernsey_postcode(self):
-        data = {"postcode": "GY1"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "The postcode GY1 is in Guernsey")
-
-    def test_invalid_guernsey_postcode(self):
-        data = {"postcode": "GY1"}
-        response = self.client.get(self.url, data)
-        self.assertContains(response, "Legal Aid in Guernsey")
-
-    def test_invalid_scottish_postcode(self):
```

Comment:
question - do we want to keep some of the invalid test cases i.e. 

- Invalid Guernsey postcode
- Invalid Scotland postcode

Reply:
Yeah, I can add one test in there.

Those current tests proved nothing unfortunately. It's the same postcode being used, it's just checking a different string on the page. 

They are also not invalid postcodes, they are valid. 

Invalid postcodes are handled by the search input field which already have tests to check for an invalid postcode. 

I will test however a bad postcode against the views to make sure it handles it fine.

Reply:
Thank you. 

I noticed a few inconsistencies when entering part of a postcode, hence the comment. e.g:

- 'BT' - does a search
- 'BT [and number] - goes to NI page.

- 'IM' - should invalid postcode error message
- 'IM [and number] - goes to Isles of Man page.

Reply:
Yeah, the postcode check still need fine tuning and there is more work to be done on it. For example TD15 is not a Scotland postcode even when we say it is.

It's out of scope of this ticket to go and fix this and the above. However bug tickets and feature tickets should be created from this. 

I've found we handled postcode not found bad requests poorly, which is something we'll also need to fix later.

Reply:
these:
'BT' - does a search

'BT [and number] - goes to NI page.

'IM' - shows invalid postcode error message

'IM [and number] - goes to Isles of Man page.
Are all due to the regex being very simple and we should probably have a ticket to address that

Reply:
I agree this ticket has highlighted the postcode issue even more.